src/js/Audit.js

```diff
@@ -218,68 +218,122 @@ axs.Audit.getRulesCannotRun = function(opt_configuration) {
+        };
+        var root = configuration.walkDom ? document.documentElement : configuration.scope;
+        // Because 'related elements' could occur anywhere in the DOM we need to start at document.documentElement
+        axs.dom.composedTreeSearch(root, null, { preorder: function(element, flags) {
+            if (!flags.inScope)
+                flags.inScope = element === configuration.scope;
+            for (var i = 0; i < auditRules.length; i++) {
+                var auditRule = auditRules[i];
+                if (!auditRule.canRun(configuration))
+                    continue;
+                auditRule._options = new AuditOptions(configuration, auditRule);  // always rebuild this, it could change each run
+                var ignore = flags.ignoring[auditRule.name] ||
+                        (flags.ignoring[auditRule.name] = auditRule._options.ignore(element));
+                if (!ignore) {
+                    auditRule.collectMatchingElement(element, flags);
+                }
+            }
+            return true;
+        } }, rootFlags);
+    };
 
-        var options = {};
+    /**
+     * Represents options pertaining to a specific AuditRule.
+     * Instances may have the following optional named parameters:
+     *     ignoreSelectors: Selectors for parts of the page to ignore for this rule.
+     *     config: Any per-rule configuration.
+     * @param {axs.AuditConfiguration} configuration Config for the audits.
+     * @param {axs.AuditRule} auditRule The AuditRule whose options we want.
+     * @constructor
+     */
+    function AuditOptions(configuration, auditRule) {
         var ignoreSelectors = configuration.getIgnoreSelectors(auditRule.name);
         if (ignoreSelectors.length > 0 || configuration.scope)
-            options['ignoreSelectors'] = ignoreSelectors;
+            this.ignoreSelectors = ignoreSelectors;
         var ruleConfig = configuration.getRuleConfig(auditRule.name);
-        if (ruleConfig != null)
-            options['config' ] = ruleConfig;
-        if (configuration.scope)
-            options['scope'] = configuration.scope;
-        if (configuration.maxResults)
-            options['maxResults'] = configuration.maxResults;
-        var result = auditRule.run.call(auditRule, options);
-        var ruleValues = axs.utils.namedValues(auditRule);
-        ruleValues.severity = configuration.getSeverity(auditRuleName) ||
-                              ruleValues.severity;
-        result.rule = ruleValues;
-        results.push(result);
+        if (ruleConfig)
+            this.config = ruleConfig;
     }
 
-    return results;
-};
-goog.exportSymbol('axs.Audit.run', axs.Audit.run);
-
+    /**
+     * Determine if this element should be ignored by the AuditRule.
+     * @param {Element} element An audit candidate.
+     * @returns {boolean} true if this element should be ignored by this AuditRule.
+     */
+    AuditOptions.prototype.ignore = function(element) {
```

Comment:
"ignore" here reads to me like the imperative; I would expect this to update the configuration to ignore the given element, based on the name.

What about something like `shouldIgnore` or `ignored` or `isIgnored`?